srcs/ResponseHandler.cpp

```diff
@@ -211,28 +198,44 @@ void ResponseHandler::handleBody(request_s& request) {
 }
 std::string& ResponseHandler::handleRequest(request_s& request) {
 	this->_response.clear();
-	if (request.method == PUT) {
+	_body.clear();
+	if (request.method == PUT)
 		handlePut(request);
-	}
-	else {
+	else
 		generateResponse(request);
-	}
+	_body.clear();
 	return _response;
 }
 
 void ResponseHandler::handlePut(request_s& request) {
 	_response = "HTTP/1.1 ";
+	struct stat statstruct = {};
 	std::string filePath = request.server->getfilepath(request.uri);
 
 	if (!request.location->checkifMethodAllowed(request.method)) {
-		request.status_code = 405;
+		this->_response += _status_codes[405];
+		handleBody(request);
+		handleALLOW(request);
+	}
+	else if (request.body.length() > request.location->getmaxbody()) { // If body length is higher than location::maxBody
+		this->_response += _status_codes[413];
+		this->_header_vals[CONNECTION] = "close";
+		handleBody(request);
+	}
+	else if (filePath[filePath.length() - 1] == '/' || (stat(filePath.c_str(), &statstruct) == 0 && S_ISDIR(statstruct.st_mode))) {
+		this->_response += _status_codes[409];
```

Comment:
I don't think 409 is suitable for these conditions. Nginx returns a 500, maybe that would be better?

Reply:
409 conflict is just fine, doesnt have to be completely nginx